submissions/jeff/bonus-testing-part-two.js

```diff
@@ -0,0 +1,89 @@
+    checking for HOW something works, but rather the end RESULT and
+    state of a particular place in our application.
+*/
+
+
+/* As a refresher, presented again is a basic example of a test
+    "framework" where our test is a function that we can reuse
+    to check against different components of our application.
+    The `!expression` may look odd - we test for failing
+    functionality first to then refactor the test to pass.
+    This is common practice in testing ('red-green' refactoring).
+*/
+function expect(expression, failureMessage, successMessage) {
+  if (!expression) {
+    console.log('test failed:', failureMessage);
+    return;
+  }
+  console.log('test passed:', successMessage);
+}
+
+/* Below is an example of our test in action. Run this file in node
+    to see what happens when it fails, and change `ricksFaveAnimal`
+    to get it to pass!
+*/
+var ricksFaveAnimal = 'hyena';
+
+expect(
+  ricksFaveAnimal === 'penguin',
+  'ricksFavoriteAnimal should equal penguin, but currently equals ' + ricksFaveAnimal,
+  'ricksFavoriteAnimal equals penguin!');
+
+  // BEGIN WORK BELOW - test code by running `node bonus-testing-part-two.js`
+  //  in your terminal!
+
+  /* ===================================================================
+  ---------------------- Part Two! Hungry Lion ----------------------------
+  ======================================================================
+   As long as the lion is well-fed, he doesn't care too much of the
+   humans that pass through. Unfortunately, the new caretaker is a little
+   absent-minded.
+
+   The lion needs 4 meals per day on average to stay happy. You're going to
+   figure out how many days it took before the lion decided to supplement his
+   diet with the caretaker.
+  */
+
+  // number of times the new caretaker fed the lion. one array entry per day
+var mealsPerDay = [5, 4, 3, 6, 2, 4, 3, 4, 5, 1];
+var tooHungryDay;
+
+  /*
+   TODO:
+   Cycle through the days in mealsPerDay. At each day, print out the average
+   number of meals/day the lion got since the new caretaker started.
+   tooHungryDay should receive the number of days before the lion started
+   pondering protein supplements (the first day the average dips below 4
+   meals)
+  */
+var averageMeals = function(array){
+  var total = 0;
+  var counter = 0;
+  var average = 0;
+  array.map(function(day){
+    total += day;
+    counter++;
+    average = total / counter;
+    console.log('Day ' + counter + ' average is: ' + average);
+    if (average < 4 && tooHungryDay === undefined) {
+      tooHungryDay = counter - 1;
+    }
+  });
+  console.log(tooHungryDay);
+  return tooHungryDay;
+};
+
+averageMeals(mealsPerDay);
+
+expect(
+  typeof(tooHungryDay) === 'number',
+  'tooHungryDay should be a number but instead is a data type of ' + typeof(tooHungryDay),
+  'The lion appears to be too hungry after ' + tooHungryDay + ' days...');
+
+  // TODO:
```

Comment:
Only thing left to do is check to make sure that our selected day is an appropriate answer based on our array.